pkg/cni/socket.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2021 Multus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package cni
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"syscall"
+)
+
+const (
+	serverSocketName                   = "multus-cni.sock"
+	fullReadWriteExecutePermissions    = 0777
+	thickPluginSocketRunDirPermissions = 0700
+)
+
+// FilesystemPreRequirements ensures the target `rundir` features the correct
+// permissions.
+func FilesystemPreRequirements(rundir string) error {
+	socketpath := SocketPath(rundir)
+	if err := os.RemoveAll(rundir); err != nil && !os.IsNotExist(err) {
+		info, err := os.Stat(rundir)
+		if err != nil {
+			return fmt.Errorf("failed to stat old pod info socket directory %s: %v", rundir, err)
+		}
+		// Owner must be root
+		tmp := info.Sys()
+		statt, ok := tmp.(*syscall.Stat_t)
```

Comment:
I will post questions as comments here whenever I do not understand something. Let me know if I should be directing these questions elsewhere. :)

Is this code guaranteed to be running on a Unix environment? Where can I find that info?

Reply:
> I will post questions as comments whenever I do not understand something. It isn't necessarily code review, so let me know if I should be directing these questions elsewhere. :)
> 
> Is this code guaranteed to be running on a Unix environment? Where can I find that info?

AFAIU yes - at least we are using netlink, which only runs on linux.

I have no idea if / where the multus project has such documentation. @dougbtv please keep me honest.